modules/apps/foundation/portal-configuration/portal-configuration-settings/src/main/java/com/liferay/portal/configuration/settings/internal/ScopedConfigurationBeanProvider.java

```diff
@@ -0,0 +1,144 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.configuration.settings.internal;
+
+import com.liferay.portal.configuration.settings.internal.util.ConfigurationPidUtil;
+import com.liferay.portal.configuration.metatype.util.ConfigurationScopedPidUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.settings.definition.ConfigurationBeanDeclaration;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.service.cm.ConfigurationEvent;
+import org.osgi.service.cm.ConfigurationListener;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+/**
+ * @author Drew Brokke
+ */
+@Component(
+	immediate = true,
+	service =
+		{ConfigurationListener.class, ScopedConfigurationBeanProvider.class}
+)
+public class ScopedConfigurationBeanProvider implements ConfigurationListener {
+
+	@Override
+	public void configurationEvent(ConfigurationEvent event) {
+		if (event.getType() != ConfigurationEvent.CM_UPDATED) {
+			return;
+		}
+
+		String pid = event.getFactoryPid();
+
+		if (pid == null) {
+			pid = event.getPid();
+		}
+
+		ScopeKey scopeKey;
+
+		try {
+			String basePid = ConfigurationScopedPidUtil.getBasePid(pid);
```

Comment:
Should we check if the pid is just a regular pid before creating the ScopeKey?
for example, `if (basePid.equals(pid)) {return;}`

I believe the ConfigurationListener listens for all configurations, both regular and scoped ones.

Reply:
All the relevant checks should happen in the constructor for the ScopeKey.  Basically, you can't make one unless it's in a valid scope, which SYSTEM isn't.